src/Blocks/BlockPiston.cpp

```diff
@@ -32,13 +32,17 @@ void cBlockPistonHandler::OnDestroyed(cChunkInterface & a_ChunkInterface, cWorld
 {
 	Vector3i blockPos(a_BlockX, a_BlockY, a_BlockZ);
 
-	// Get the extension of the piston
```

Comment:
This comment is no longer valid, I think a more general one would be better:
`If the piston is extended, destroy the extension as well`
And get rid of the empty line between setting `OldMeta` and testing it.